pkg/guacanalytics/patchPlanning.go

```diff
@@ -190,3 +200,26 @@ func exploreIsDependencyFromDepPkg(ctx context.Context, gqlclient graphql.Client
 
 	return nil
 }
+
+// TODO: impelement this functions
+func exploreIsOccurrence(ctx context.Context, gqlclient graphql.Client, q *queueValues, isOccurrence model.NeighborsNeighborsIsOccurrence) error {
+	// Step 1: Find Artifact attached to package through IsOccurence
+	// -> call .Artifact on isOccurrence
+	// Step 2: Find HasSLSA where Artifact is the builtFrom
+	// -> call .Neighbors on the artifact id with the edge type specified as HasSLSAArtifact and loop through results
```

Comment:
Calling `.Neighbors` and acting upon it sounds a lot like the main loop, what is the special case here?

Reply:
The main loop found the IsOccurrence (say this is called A) attached to the original package or source but here we need to call .Neighbors twice more to 
1) Find the HasSLSA connected to the artifact connected to A
2) Find the IsOccurrence connected to the Artifact that is the subject of the HasSLSA

In short, there is a longer chain to traverse in the case of an IsOccurrence to reach the relevent package affected.

It would look something like this:

Package (or Source) ----> isOccurrence (A) ----> Artifact ----> HasSLSA (B) ----> Artifact (the subject of the SLSA) ----> isOccurrence (C) ----> Package (or Source)

The first .Neighbors is for isOccurrence A where the second/third .Neighbors are for B and C.  I was planning to do this in order to get from the original Package (or Source) to the Package (or Source) at the end of the chain that needs to be added to the queue.

Reply:
How about the case of `PkgEqual`, would `PkgEqual` also explore `IsDepedency`?

Reply:
I haven't planned out or taken into account in the code a PkgEqual case but in theory it should